rust/agents/validator/src/submit.rs

```diff
@@ -58,17 +58,22 @@ impl ValidatorSubmitter {
         }
     }
 
+    /// Submits signed checkpoints.
+    /// If a target checkpoint is provided, it must not be bethind the provided tree, and
+    /// the submitter will only submit checkpoints until it reaches the target checkpoint.
+    /// If a target checkpoint is not provided, the submitter will submit checkpoints indefinitely.
     #[instrument(err, skip(self, tree), fields(domain=%self.merkle_tree_hook.domain()))]
     pub(crate) async fn checkpoint_submitter(
         self,
         mut tree: IncrementalMerkle,
         target_checkpoint: Option<Checkpoint>,
     ) -> Result<()> {
-        let mut checkpoint_queue = vec![];
-
-        let mut reached_target = false;
+        // If the target checkpoint is provided, the provided tree must not be behind it.
+        if let Some(target) = target_checkpoint {
```

Comment:
Isn't this mostly the case for the backfill submitter at startu[? I.e. the tree is empty, but the target is the current checkpoint on-chain?

Reply:
Yeah this was just to be safe without assuming how it's called, though now with the fn split this can go away